app/renderer/components/Inspector/LocatorTestModal.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import { Modal, Input, Select, Row, Col, Button } from 'antd';
+import InspectorStyles from './Inspector.css';
+
+const { Option } = Select;
+
+export default class LocatorTestModal extends Component {
+
+  onSubmit () {
+    const {locatedElements, locatorTestStrategy, locatorTestValue, searchForElement, clearSearchResults, hideLocatorTestModal} = this.props;
+    if (locatedElements) {
+      hideLocatorTestModal();
+      clearSearchResults();
+    } else {
+      searchForElement(locatorTestStrategy, locatorTestValue);
+    }
+  }
+
+  onCancel () {
+    const {hideLocatorTestModal, clearSearchResults} = this.props;
+    hideLocatorTestModal();
+    clearSearchResults();
+  }
+
+  render () {
+    const {isLocatorTestModalVisible, isSearchingForElements, locatedElements, applyClientMethod,
+      setLocatorTestValue, locatorTestValue, setLocatorTestStrategy, locatorTestStrategy, setLocatorTestElement, locatorTestElement, clearSearchResults} = this.props;
+
+    const locatorStrategies = [
+      ['id', 'Id'],
+      ['xpath', 'XPath'],
+      ['name', 'Name'],
+      ['class name', 'Class Name'],
+      ['accessibility id', 'Accessibility ID'],
+      ['-android uiautomator', 'UIAutomator Selector (Android)'],
+      ['-ios predicate string', 'Predicate String (iOS)'],
+      ['-ios class chain', 'Class Chain (iOS)'],
+    ];
+
+    return <Modal visible={isLocatorTestModalVisible} 
+      okText={locatedElements ? 'Done' : 'Search'}
+      cancelText='Cancel'
+      title='Search for element'
+      confirmLoading={isSearchingForElements}
+      onOk={(() => this.onSubmit()).bind(this)}
+      onCancel={(() => this.onCancel()).bind(this)}>
+        {!locatedElements && <Row>
```

Comment:
we've got a few different sections here, each annotated with `locatedElements` or `!locatedElements`. it's fine but the pattern suggests maybe these are actually two different components?